### PR TITLE
executor: reduce a comparison in union scan

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: executor/union_scan.go [hunk 1]
```diff
@@ -195,10 +177,13 @@ func (us *UnionScanExec) getAddedRow() Row {
 	return addedRow
 }
 
-func (us *UnionScanExec) pickRow(a, b Row) (Row, error) {
+// pickRow picks the row in order.
+// The second row retrund is whether picks the first input row.
```
Comment (hanfei1991) [inline executor/union_scan.go]: a typo : returned

Comment (coocood) [inline executor/union_scan.go]: Please refine the comment.

**Comment by zimulala** [comment]
PTAL @coocood @hanfei1991 @winoros

File: executor/union_scan.go [hunk 2]
```diff
@@ -195,10 +177,13 @@ func (us *UnionScanExec) getAddedRow() Row {
 	return addedRow
 }
 
-func (us *UnionScanExec) pickRow(a, b Row) (Row, error) {
+// pickRow picks the row in order.
+// The second value retruned is used to determine whether to return the first input row.
```
Comment (coocood) [inline executor/union_scan.go]: s/retruned/returned

File: executor/union_scan.go [hunk 3]
```diff
@@ -195,10 +177,13 @@ func (us *UnionScanExec) getAddedRow() Row {
...
+func (us *UnionScanExec) pickRow(a, b Row) (Row, bool, error) {
```
Comment (coocood) [inline executor/union_scan.go]: If we return if the first is picked, then we don't need to return the row.

**Comment by zimulala** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Pull request merged**